File: 0961-n-repeated-element-in-size-2n-array/0961-n-repeated-element-in-size-2n-array.cpp
class Solution {
public:
    int repeatedNTimes(vector<int>& nums) {
        int len=nums.size()/2;
        unordered_map<int,int> m;
        
        for(int i=0;i<nums.size();i++){
            m[nums[i]]++;
        }
        
        for(auto x:m){
            if(x.second==len)
                return x.first;
        }
        return 0;
    }
};

File: 0961-n-repeated-element-in-size-2n-array/0961-n-repeated-element-in-size-2n-array.java
class Solution {
    public int repeatedNTimes(int[] nums) {
        int n=nums.length;
        Set<Integer> st = new HashSet<>();

        for (int i = 0; i < n; i++) {
            if (st.contains(nums[i]))
                return nums[i];
            st.add(nums[i]);
        }

        return -1;
    }
}